app/javascript/selectors/errorsSelectors.js

```
@@ -1,31 +1,48 @@
 )
-export const getTotalScreeningSubmissionErrorValueSelector = createSelector(
+export const getConstraintValidationErrorsSelector = createSelector(
   getErrors,
-  (errors) => {
-    if (errors.has(SUBMIT_SCREENING_COMPLETE)) {
-      return errors.get(SUBMIT_SCREENING_COMPLETE).size
-    } else {
-      return 0
-    }
-  }
+  (errors) => (errors.toIndexedSeq().flatten(true).filter((error) => (error && error.get('type') === 'constraint_validation')).toJS())
 )
 export const getScreeningSubmissionErrorsSelector = createSelector(
+  getConstraintValidationErrorsSelector,
+  (constraintValidationErrors) => (
+    constraintValidationErrors.map((error) =>
+      (`${error.property} ${error.user_message} (Incident Id: ${error.incident_id})`))
+  )
+)
+export const getSystemErrorsSelector = createSelector(
   getErrors,
-  (errors) => {
-    if (errors.has(SUBMIT_SCREENING_COMPLETE)) {
-      return errors.get(SUBMIT_SCREENING_COMPLETE).map((error) =>
-        (`${error.get('property')} ${error.get('user_message')} (Incident Id: ${error.get('incident_id')})`)
-      )
+  (errors) => (errors.toIndexedSeq().flatten(true).filter((error) => (error && error.get('type') !== 'constraint_validation')).toJS())
+)
+export const getSystemErrorIncidentIdsSelector = createSelector(
+  getSystemErrorsSelector,
+  (systemErrors) => (systemErrors.map((error) => {
+    if (error) {
+      return error.incident_id
+    }
+    return ''
+  }).join(', '))
+)
+export const getPageErrorMessageSelector = createSelector(
+  getErrors,
+  getSystemErrorIncidentIdsSelector,
+  getConstraintValidationErrorsSelector,
+  (errors, systemErrorIncidentIds, constraintValidationErrors) => {
+    if (constraintValidationErrors.length > 0) {
+      return `${constraintValidationErrors.length} error(s) have been identified. Please fix them and try submitting again.`
     } else {
-      return List()
+      let message = 'Something went wrong, sorry! Please try your last action again.'
```

Since the PageError component will show this message by default, could we simplify this else to be else if systemErrorIncidentIds and return null or undefined if false so the component will use the default message when no id is present?

---

I thought about this, but it's much cleaner to explicitly return the error message instead of composing it with conditional logic.